src/Microsoft.DotNet.CodeFormatter.Analyzers/ProvideExplicitVariableTypeFixer.cs

```diff
@@ -84,7 +84,15 @@ public override FixAllProvider GetFixAllProvider()
                                           .WithAdditionalAnnotations(Simplifier.Annotation)
                                           .WithTriviaFrom(varNode);
             documentEditor.ReplaceNode(varNode, explicitTypeNode);
-            return documentEditor.GetChangedDocument();
+            var newDocument = documentEditor.GetChangedDocument();
+
+            // We don't want the explicit type to be fully qualified. So add an using for this node and the simplifier will
+            // take care of removing it if it isn't necessary.
+            // The second parmaeter to AddImportsAsync is an annotation that is used to locate the span
+            // for which an using should be added. Since we added the simplify annotation on explicitTypeNode, 
+            // we can just use that annotation to locate the node again in newDocument. 
```

Comment:
If I understand this right, the following call says: to the specified document, add any imports that are required by the node with the "simplifier" annotation. And then later on, the fact that this node _has_ the simplifier annotation will cause the fully qualified type name to be replaced with a simple type name.

Can this go wrong in any way? Is it possible that some node could have a simplifier annotation without us having put it there? If so, we could invent our own "provide explicit variable type" annotation, and add that to the node _in addition to_ the simplifier annotation. Then here, we could pass this annotation to `AddImportAsync` (instead of passing the simplifier annotation), secure in the knowledge that the only nodes with this annotation are the ones we explicitly marked ourselves?